modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/CriteriaGroup.es.js

```diff
@@ -112,16 +113,37 @@ class CriteriaGroup extends Component {
 	 * @memberof CriteriaGroup
 	 */
 	_handleCriterionAdd = (index, criterion) => {
-		const {criteria, onChange, root, supportedOperators} = this.props;
+		const {
+			criteria,
+			onChange,
+			root,
+			supportedOperators,
+			supportedPropertyTypes
+		} = this.props;
+
+		const {
+			defaultValue,
+			operatorName,
+			propertyName,
+			type,
+			value
+		} = criterion;
 
-		const {defaultValue: value, operatorName, propertyName} = criterion;
+		const criterionValue = value || defaultValue;
```

Comment:
Would it make more sense to do something like this above?

```js
		const {
			defaultValue,
			operatorName,
			propertyName,
			type,
			value = defaultValue
		} = criterion;
```

Reply:
@bryceosterhaus Both `value` and `defaultValue` are being destructured from `criterion` and we can't set the default `value` with another value being destructured together.

```javascript
const test = {color: 'blue'};
const {value = color} = test; // Uncaught ReferenceError: color is not defined
```

Reply:
Oh I guess this will work? I don't know if this would be strange though.

```javascript
const test = {color: 'blue'};

const {
   value = test.color
} = test;
```

Reply:
Oh yeah good point, I didnt realize it was coming from the criterion object as well. I agree its more readable how you have it now